### PR TITLE
added tests; refactor some code.

### PR DESCRIPTION
**Pull request by corescan**

### CODE REVIEW

File: src/Select.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ import SelectOption from './SelectOption';
 import OutsideClick from 'ship-components-outsideclick';
 import HighlightClick from 'ship-components-highlight-click';
 
-import dispatch from './dispatch';
+import dispatch from './lib/dispatch';
```
Comment (isuttell) [inline src/Select.js]: Hey, did you test this in any of the apps? I'm not sure the regex we use in webpack will babelify files in sub folders of the apps.

File: src/Select.js [hunk 2]
```diff
@@ -70,7 +71,7 @@ export default class Select extends React.Component {
    */
   componentDidUpdate(prevProps, prevState) {
     if (this.props.options.length > 5 && this.refs.selected && prevState.active === false && this.state.active === true) {
-      this.refs.selected.scrollIntoView();
+      scrollIntoView(ReactDOM.findDOMNode(this.refs.selected));
```
Comment (isuttell) [inline src/Select.js]: Do we need ReactDOM here?

File: src/__tests__/Select-spec.jsx [hunk 3]
```diff
@@ -1,39 +1,139 @@
...
+        options: options
+       }
     );
 
     expect(() => TestUtils.renderIntoDocument(element))
        .not.toThrow();
   });
 
-  it('should support custom css classes', () => {
+  it('defines refs to input and list nodes', () => {
```
Comment (isuttell) [inline src/__tests__/Select-spec.jsx]: What does testing if the refs get set do? What's the functionality here you are testing?

File: src/Select.js (hunk 2 above)
Comment (corescan) [reply src/Select.js]: Yes, in this case "scrollIntoView" is a method of the SelectOption Component class. So, in short, when I separated scrollIntoView into a lib folder it became a static function that requires a DOM node.

File: src/__tests__/Select-spec.jsx (hunk 3 above)
Comment (corescan) [reply src/__tests__/Select-spec.jsx]: Good point. Other tests that assert functionality should implicitly handle this case.

**Comment by corescan** [comment]
No idea why coveralls rating decreased. I moved a couple functions into their own files under `/lib` and according to Jest's coverage report, I increased the coverage a good bit. Clearly everything bombed here with Travis, but locally everything passed.

File: src/Select.js (hunk 1 above)
Comment (corescan) [reply src/Select.js]: I did not. I just tested the examples page which seems to be working.

Comment (corescan) [reply src/Select.js]: Verified to be working in OneSony against this webpack loader regex: `/ship-components-.*\/src/`

Should be ok to merge up now.

**Pull request merged**